services/wallet/persistence.go

```diff
@@ -33,9 +33,6 @@ func (p *Persistence) SaveTokens(tokens map[common.Address][]Token) (err error)
 	for address, addressTokens := range tokens {
 		for _, t := range addressTokens {
 			for chainID, b := range t.BalancesPerChain {
-				if b.HasError || b.Balance.Cmp(big.NewFloat(0)) == 0 {
```

Comment:
That part doesn't make sense to me because balances which become zero will not be updated. Obviously it would cause problems with cache usage. `INSERT` statement is used later but the primary key has `ON CONFLICT REPLACE`, so that's effectively an update.

Reply:
yes you are right